### PR TITLE
[fix](memory) Remind log if `vm/overcommit_memory`=2 when be start

### PR DESCRIPTION
**Pull request by xinyiZzz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Expect vm overcommit memory value to be 1,  system will no longer throw bad_alloc, memory alloc are always accepted,
memory limit check is handed over to Doris Allocator, make sure throw exception position is controllable,
otherwise bad_alloc can be thrown anywhere and it will be difficult to achieve exception safety.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: bin/start_be.sh [hunk 1]
```diff
@@ -20,6 +20,13 @@ set -eo pipefail
 
 curdir="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
 
+if [ $(id -u) -eq 0 ]; then # current user has root privileges
+    # system will no longer throw bad_alloc, memory alloc are always accepted,
```
Comment (yiguolei) [inline bin/start_be.sh:24]: maybe we should not do that. it's better to print a warning? or add the message to be.out?
but i am not sure.

Comment (xinyiZzz) [reply bin/start_be.sh]: You are right, should not directly modify sys conf
Changed to print log

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/util/mem_info.cpp [hunk 2]
```diff
@@ -300,13 +300,28 @@ void MemInfo::init() {
         _s_sys_mem_available_warning_water_mark = _s_sys_mem_available_low_water_mark + p1;
     }
 
+    // Expect vm overcommit memory value to be 1, system will no longer throw bad_alloc, memory alloc are always accepted,
+    // memory limit check is handed over to Doris Allocator, make sure throw exception position is controllable,
+    // otherwise bad_alloc can be thrown anywhere and it will be difficult to achieve exception safety.
+    std::ifstream sys_vm("/proc/sys/vm/overcommit_memory", std::ios::in);
```
Comment (yiguolei) [inline be/src/util/mem_info.cpp:306]: Add #if os_linux macro to do this check only in linux not macos.

Comment (xinyiZzz) [reply be/src/util/mem_info.cpp:306]: Now in #ifndef __APPLE__

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**